### ISSUE TITLE
linter: `no-useless-assignment` false positive

### ISSUE THREAD

**Issue by KieranP** [oxc-project/oxc#21359]
### What version of Oxlint are you using?

1.59.0

### What command did you run?

oxlint -c

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint'

export default defineConfig({
  ignorePatterns: ['/node_modules', '/*.{js,ts}'],
  plugins: ['eslint', 'oxc', 'typescript', 'import', 'promise', 'unicorn'],
  categories: {
    correctness: 'error',
    nursery: 'error',
    pedantic: 'error',
    perf: 'error',
    restriction: 'error',
    style: 'error',
    suspicious: 'error',
  },
  env: {
    builtin: true,
    browser: true,
  },
})
```

### What happened?

False positive on no-useless-assignment when assigning variable in a loop followed by continue call

```ts
function anyChanges(before: object, after: object) {
  let hasChanges = false

  // Check for added keys
  for (const key of Object.keys(after)) {
    if (!(key in before)) {
      hasChanges = true
      continue
    }

    // More logic
  }

  // Check for deleted keys
  for (const key of Object.keys(before)) {
    if (!(key in after)) {
      hasChanges = true
      continue
    }

    // More logic
  }

  return hasChanges
}
```

```
× [eslint(no-useless-assignment)](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-useless-assignment.html): This assigned value is not used in subsequent statements.
    ╭─[app/frontend/javascripts/helpers/object_diff.ts:11:7]
 10 │     if (!(key in before)) {
 11 │       hasChanges = true
    ·       ──────────
 12 │       continue
    ╰────
  help: Consider removing or reusing the assigned value.

× [eslint(no-useless-assignment)](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-useless-assignment.html): This assigned value is not used in subsequent statements.
    ╭─[app/frontend/javascripts/helpers/object_diff.ts:21:7]
 20 │     if (!(key in after)) {
 21 │       hasChanges = true
    ·       ──────────
 22 │       continue
    ╰────
  help: Consider removing or reusing the assigned value.
```

**Comment by camc314** [comment]
fixed by https://github.com/oxc-project/oxc/pull/21358
